src/Compilers/VisualBasic/Portable/Emit/SymbolAdapter.vb

```diff
@@ -36,39 +36,37 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim synthesized As ArrayBuilder(Of SynthesizedAttributeData) = Nothing
             AddSynthesizedAttributes(compilationState, synthesized)
-            Return GetCustomAttributesToEmit(Me.GetAttributes(), synthesized, isReturnType:=False, emittingAssemblyAttributesInNetModule:=emittingAssemblyAttributesInNetModule)
+
+            Return GetCustomAttributesToEmit(Me.GetAttributes(), synthesized,
+                                             Function(attribute, symbol) attribute.ShouldEmitAttribute(symbol, isReturnType:=False, emittingAssemblyAttributesInNetModule:=False))
         End Function
 
         ''' <summary> 
         ''' Returns a list of attributes to emit to CustomAttribute table.
         ''' </summary>
         Friend Function GetCustomAttributesToEmit(userDefined As ImmutableArray(Of VisualBasicAttributeData),
                                                   synthesized As ArrayBuilder(Of SynthesizedAttributeData),
-                                                  isReturnType As Boolean,
-                                                  emittingAssemblyAttributesInNetModule As Boolean) As IEnumerable(Of VisualBasicAttributeData)
+                                                  shouldEmitAttribute As Func(Of VisualBasicAttributeData, Symbol, Boolean)) As IEnumerable(Of VisualBasicAttributeData)
 
             ' PERF: Avoid creating an iterator for the common case of no attributes.
             If userDefined.IsEmpty AndAlso synthesized Is Nothing Then
                 Return SpecializedCollections.EmptyEnumerable(Of VisualBasicAttributeData)()
             End If
 
-            Return GetCustomAttributesToEmitIterator(userDefined, synthesized, isReturnType, emittingAssemblyAttributesInNetModule)
+            Return GetCustomAttributesToEmitIterator(userDefined, synthesized, shouldEmitAttribute)
         End Function
 
         Private Iterator Function GetCustomAttributesToEmitIterator(userDefined As ImmutableArray(Of VisualBasicAttributeData),
                                                   synthesized As ArrayBuilder(Of SynthesizedAttributeData),
-                                                  isReturnType As Boolean,
-                                                  emittingAssemblyAttributesInNetModule As Boolean) As IEnumerable(Of VisualBasicAttributeData)
+                                                  shouldEmitAttribute As Func(Of VisualBasicAttributeData, Symbol, Boolean)) As IEnumerable(Of VisualBasicAttributeData)
 
             If synthesized IsNot Nothing Then
                 For Each attribute In synthesized
-                    Debug.Assert(attribute.ShouldEmitAttribute(Me, isReturnType, emittingAssemblyAttributesInNetModule:=False))
+                    Debug.Assert(shouldEmitAttribute(attribute, Me))
```

Comment:
Point of interest: `emittingAssemblyAttributesInNetModule` used to be hard-coded to false (in VB, but not in corresponding C# code). I think both this check and the one below should match.